### PR TITLE
fix javadocs for ok http module

### PR DESCRIPTION
**Pull request by uaihebert**
#231 first part of javadocs fixes

I am proposing to split this ticket #231 into smaller PRs. it will make it easier to review.

### CODE REVIEW

**Comment by breedx-nr** [comment]
That is a very good idea to split it up into one for each module.  I'll make new issues for the other modules.

File: telemetry-http-okhttp/src/main/java/com/newrelic/telemetry/OkHttpPoster.java [hunk 1]
```diff
@@ -25,12 +25,20 @@ public OkHttpPoster() {
     this(Duration.ofSeconds(2));
   }
 
-  /** Create an OkHttpPoster with a default OkHttpClient, with only a custom call timeout. */
+  /**
+   * Create an OkHttpPoster with a default OkHttpClient, with only a custom http timeout.
+   *
+   * @param callTimeout expected connection timeout
```
Comment (breedx-nr) [inline telemetry-http-okhttp/src/main/java/com/newrelic/telemetry/OkHttpPoster.java:31]: Let's change this from `connection timeout` to `call timeout`.  The javadocs for OkHttp explain that "* The call timeout spans the entire call: resolving DNS, connecting, writing the request body, server processing, and reading the response body.", so it's more than just connect.

File: telemetry-http-okhttp/src/main/java/com/newrelic/telemetry/SimpleMetricBatchSender.java [hunk 2]
```diff
@@ -51,6 +55,9 @@ public static MetricBatchSenderBuilder builder(String apiKey) {
    * @see <a
    *     href="https://docs.newrelic.com/docs/apis/getting-started/intro-apis/understand-new-relic-api-keys#user-api-key">New
    *     Relic API Keys</a>
+   * @param apiKey new relic api key
+   * @param callTimeout http connection timeout
```
Comment (breedx-nr) [inline telemetry-http-okhttp/src/main/java/com/newrelic/telemetry/SimpleMetricBatchSender.java:59]: `call timeout` please.

File: telemetry-http-okhttp/src/main/java/com/newrelic/telemetry/SimpleMetricBatchSender.java [hunk 3]
```diff
@@ -63,6 +70,9 @@ public static MetricBatchSender build(String apiKey, Duration callTimeout) {
...
```
Comment (breedx-nr) [inline telemetry-http-okhttp/src/main/java/com/newrelic/telemetry/SimpleMetricBatchSender.java:74]: call timeout

File: telemetry-http-okhttp/src/main/java/com/newrelic/telemetry/SimpleSpanBatchSender.java [hunk 4]
```diff
@@ -51,6 +55,9 @@ public static SpanBatchSenderBuilder builder(String apiKey) {
    * @see <a
    *     href="https://docs.newrelic.com/docs/apis/getting-started/intro-apis/understand-new-relic-api-keys#user-api-key">New
    *     Relic API Keys</a>
+   * @param apiKey new relic api key
+   * @param callTimeout http connection timeout
```
Comment (breedx-nr) [inline telemetry-http-okhttp/src/main/java/com/newrelic/telemetry/SimpleSpanBatchSender.java:59]: call timeout

File: telemetry-http-okhttp/src/main/java/com/newrelic/telemetry/SimpleSpanBatchSender.java [hunk 5]
```diff
@@ -63,6 +70,9 @@ public static SpanBatchSender build(String apiKey, Duration callTimeout) {
...
```
Comment (breedx-nr) [inline telemetry-http-okhttp/src/main/java/com/newrelic/telemetry/SimpleSpanBatchSender.java:74]: call timeout instead of connection timeout

**Review by breedx-nr — CHANGES_REQUESTED** [review]
Thanks for helping out with this!  I asked for a small change so that users don't think it's just a connection timeout when it's a timeout for the full call.  Looks great otherwise.

**Comment by breedx-nr** [comment]
I created split issues with #235 #236 and #237 .

@uaihebert Let me know if you want assignments on any of those.  🥂

**Comment by breedx-nr** [comment]
I think you have to comment on them before I can assign outside contributors.

**Comment by uaihebert** [comment]
Oh, that is new to me. I will do it. Just a second.

**Comment by uaihebert** [comment]
just pushed the updates. let me know if that is ok. 

when we finish that, I will start with the others.

**Review by breedx-nr — APPROVED** [review]

**Comment by breedx-nr** [comment]
Excellent!  Thanks again.

**Pull request merged**